internal/provider/namespaces_datasource.go

```diff
@@ -0,0 +1,445 @@
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"common_name": schema.StringAttribute{
+										Computed:    true,
+										Description: "The certificate's common name.",
+									},
+									"organization": schema.StringAttribute{
+										Computed:    true,
+										Description: "The certificate's organization.",
+									},
+									"organizational_unit": schema.StringAttribute{
+										Computed:    true,
+										Description: "The certificate's organizational unit.",
+									},
+									"subject_alternative_name": schema.StringAttribute{
+										Computed:    true,
+										Description: "The certificate's subject alternative name (or SAN).",
+									},
+								},
+							},
+						},
+						"codec_server": schema.SingleNestedAttribute{
+							Optional:    true,
+							Computed:    true,
+							Description: "A codec server is used by the Temporal Cloud UI to decode payloads for all users interacting with this namespace, even if the workflow history itself is encrypted.",
+							Attributes: map[string]schema.Attribute{
+								"endpoint": schema.StringAttribute{
+									Computed:    true,
+									Description: "The endpoint of the codec server.",
+								},
+								"pass_access_token": schema.BoolAttribute{
+									Computed:    true,
+									Description: "If true, Temporal Cloud will pass the access token to the codec server upon each request.",
+								},
+								"include_cross_origin_credentials": schema.BoolAttribute{
+									Computed:    true,
+									Description: "If true, Temporal Cloud will include cross-origin credentials in requests to the codec server.",
+								},
+							},
+						},
+						"endpoints": schema.SingleNestedAttribute{
+							Computed:    true,
+							Description: "The endpoints for the namespace.",
+							Attributes: map[string]schema.Attribute{
+								"web_address": schema.StringAttribute{
+									Description: "The web ui address.",
+									Computed:    true,
+								},
+								"grpc_address": schema.StringAttribute{
+									Computed:    true,
+									Description: "The grpc hostport address that the temporal workers, clients and tctl connect to.",
+								},
+							},
+						},
+						"private_connectivities": schema.ListNestedAttribute{
+							Optional:    true,
+							Computed:    true,
+							Description: "The private connectivities for the namespace, if any.",
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"region": schema.StringAttribute{
+										Computed:    true,
+										Description: "The id of the region where the private connectivity applies.",
+									},
+									"aws_private_link_info": schema.SingleNestedAttribute{
+										Computed:    true,
+										Optional:    true,
+										Description: "The AWS PrivateLink info. This will only be set for an AWS region.",
```

Comment:
It took me a second to parse this but now I think I understand - what about something like: "This will only be set for namespaces whose cloud provider is AWS"?

Reply:
Admittedly this comment was copy-pasted from the cloud service proto model 😅 Your wording is much clearer, I'll update both the proto and the schema description rn, and open a PR in the cloud service proto repo when I get the chance.